lsp/nls/src/linearization.rs

```diff
@@ -124,14 +133,56 @@ impl Linearizer<BuildingResource, (UnifTable, HashMap<usize, Ident>)> for Analys
                 ty,
                 kind: TermKind::Record(attrs.clone()),
                 scope: self.scope.clone(),
+                meta: self.meta.take(),
             }),
 
+            Term::MetaValue(meta) => {
+                // Notice 1: No push to lin
+                // Notice 2: we discard the encoded value as anything we
+                //           would do with the value will be handled in the following
+                //           call to [Self::add_term]
+                let meta = MetaValue {
+                    value: None,
+                    ..meta.to_owned()
+                };
+
+                for contract in meta.contracts.iter().cloned() {
```

Comment:
Do we really need to clone the iterator here? It feels like we are only matching on stuff, or at worst need to clone identifiers, but not the whole contracts.

Reply:
No, fixed in #447